### PR TITLE
Replacing Objects.equal with ==

### PR DESCRIPTION
**Pull request by bapan25**
Signed-off-by: lazycoderextreme <ruptam25@gmail.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
#1116

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Review by miovd — APPROVED** [review]

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/util/IidmXmlUtil.java [hunk 1]
```diff
@@ -326,7 +326,7 @@ public static void writeDoubleAttributeFromMinimumVersion(String rootElementName
      */
     public static void writeDoubleAttributeFromMinimumVersion(String rootElementName, String attributeName, double value, double defaultValue,
                                                               ErrorMessage type, IidmXmlVersion minVersion, NetworkXmlWriterContext context) {
-        writeAttributeFromMinimumVersion(rootElementName, attributeName, !Objects.equals(value, defaultValue), type,
+        writeAttributeFromMinimumVersion(rootElementName, attributeName, value != defaultValue, type,
```
Comment (miovd) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/util/IidmXmlUtil.java:329]: Should not be changed for `double`

**Review by miovd — CHANGES_REQUESTED** [review]

**Comment by geofjamg** [comment]
what would be the impact on network store as we most of the time recreate subobjects at each access?

**Comment by miovd** [comment]
> what would be the impact on network store as we most of the time recreate subobjects at each access?

@geofjamg Do you override the `equals` method in network-store implementation? If yes, this would indeed create issues...

**Comment by geofjamg** [comment]
> > what would be the impact on network store as we most of the time recreate subobjects at each access?
> 
> @geofjamg Do you override the `equals` method in network-store implementation? If yes, this would indeed create issues...

No, indeed should be ok woth network store

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/util/IidmXmlUtil.java (hunk 1 above)
Comment (bapan25) [reply iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/util/IidmXmlUtil.java]: done

**Review by miovd — APPROVED** [review]

**Pull request merged**